src/lib/voiceSearchHelper/index.ts

```diff
@@ -131,6 +134,14 @@ export default function createVoiceSearchHelper({
     recognition = undefined;
   };
 
+  const stop = (): void => {
+    dispose();
+    // Because `dispose` removes event listeners, `end` listener is not called.
+    // So we're setting the `status` as `finished` here.
+    // If we don't do it, it will be still `waiting` or `recognizing`.
+    resetState('finished');
```

Comment:
Can we add a test to check the status at this point?

Reply:
Good idea. Thanks!
851de97